scripts/deploy_cluster_tar_single.sh

```diff
@@ -0,0 +1,174 @@
+#
+# Licensed under the Apache License, Version 2.0 (the "License").
+# You may not use this file except in compliance with the License.
+# A copy of the License is located at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# or in the "license" file accompanying this file. This file is distributed
+# on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+# express or implied. See the License for the specific language governing
+# permissions and limitations under the License.
+
+# This script is to automate the single cluster deployment process of OpenSearch and OpenSearch-Dashboards
+
+set -e
+CURR_DIR=`pwd`
+ROOT=`dirname $(realpath $0)`; echo $ROOT; cd $ROOT
+
+function usage() {
+    echo ""
+    echo "This script is used to deploy OpenSearch and OpenSearch-Dashboards single node cluster. It downloads the latest artifacts or specified ones per user, extract them, and deploy to the localhost to start the cluster."
+    echo "--------------------------------------------------------------------------"
+    echo "Usage: $0 [args]"
+    echo ""
+    echo "Required arguments:"
+    echo -e "-v VERSION\t(1.0.0 | 1.0.0-beta1 | etc.) Specify the OpenSearch version number that you are building. This will be used to download the artifacts."
+    echo -e "-t TYPE\t(snapshots | releases) Specify the OpenSearch Type of artifacts to use, snapshots or releases."
+    echo ""
+    echo "Optional arguments:"
+    echo -e "-c \tCleanup Existing deployment only without new deployment."
+    echo -e "-s ENABLE_SECURITY\t(true | false) Specify whether you want to enable security plugin or not. Default to true."
+    echo -e "-h\t\tPrint this message."
+    echo "--------------------------------------------------------------------------"
+}
+
+function cleanup() {
+    echo ""
+    echo Clean Up
+    echo Kill Existing OpenSearch/Dashboards Process
+    (kill -9 `ps -ef | grep -i [o]pensearch | awk '{print $2}'` > /dev/null 2>&1) || echo -e "\tClear OpenSearch Process"
+    (kill -9 `ps -ef | grep -i [n]ode | awk '{print $2}'` > /dev/null 2>&1) || echo -e "\tClear Dashboards Process"
+    (kill -9 `ps -ef | grep -i [p]erformance | awk '{print $2}'` > /dev/null 2>&1) || echo -e "\tClear PerformanceAnalyzer Process"
+
+    echo Check PID List
+    (ps -ef | grep -i [o]pensearch) || echo -e "\tNo OpenSearch PIDs"
+    (ps -ef | grep -i [n]ode) || echo -e "\tNo Dashboards PIDs"
+    (ps -ef | grep -i [p]erformance) || echo -e "\tNo PerformanceAnalyzer PIDs"
+
+    echo Remove Old Deployments
+    if [ -z "$TMPDIR" ]
+    then
+      rm -rf /tmp/*_INTEGTEST_WORKSPACE
+    else
+      rm -rf $TMPDIR/*_INTEGTEST_WORKSPACE
+    fi
+}
+
+
+while getopts ":hct:v:s:p:" arg; do
+    case $arg in
+        v)
+            VERSION=$OPTARG
+            ;;
+        c)
+            cleanup
+            exit
+            ;;
+        t)
+            TYPE=$OPTARG
+            ;;
+        s)
+            ENABLE_SECURITY=$OPTARG
+            ;;
```

Comment:
Where is ENABLE_SECURITY used? I don't see any other references to it in this PR.

Reply:
We do not have security disable param yet, so please do not approve it since it is still in progress.

Reply:
Can you pull it out of this PR then? Make a new PR with the security stuff once this one is approved.

Reply:
Removed.